pkg/destroy/azure/azure.go

```diff
@@ -226,9 +247,95 @@ func (o *ClusterUninstaller) Run() (*types.ClusterQuota, error) {
 		o.Logger.Debug(err)
 	}
 
+	deadline, _ = waitCtx.Deadline()
+	diff = time.Until(deadline)
+	if diff > 0 {
+		waitCtx, _ = context.WithTimeout(context.Background(), diff) //nolint: govet ctx is thru its parent
+	}
+
+	if err := removeSharedTags(
+		waitCtx, o.resourceGraphClient, o.tagsClient, o.InfraID, o.SubscriptionID, o.Logger,
+	); err != nil {
+		errs = append(errs, fmt.Errorf("failed to remove shared tags: %w", err))
+		o.Logger.Debug(err)
+	}
+
 	return nil, utilerrors.NewAggregate(errs)
 }
 
+func removeSharedTags(
+	ctx context.Context,
+	graphClient *armresourcegraph.Client,
+	tagsClient *armresources.TagsClient,
+	infraID, subscriptionID string,
+	logger logrus.FieldLogger,
+) error {
+	tagKey := fmt.Sprintf("kubernetes.io_cluster.%s", infraID)
+	query := fmt.Sprintf(
+		"resources | where tags.['%s'] == 'shared' | project id, name, type",
+		tagKey,
+	)
+	results, err := graphClient.Resources(ctx,
+		armresourcegraph.QueryRequest{
+			Query: &query,
+			Subscriptions: []*string{
+				&subscriptionID,
+			},
+			Options: &armresourcegraph.QueryRequestOptions{
+				ResultFormat: azcoreto.Ptr(armresourcegraph.ResultFormatObjectArray),
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to query resources with shared tag: %w", err)
+	}
+
+	tagsParam := armresources.TagsPatchResource{
+		Operation: azcoreto.Ptr(armresources.TagsPatchOperationDelete),
+		Properties: &armresources.Tags{
+			Tags: map[string]*string{
+				tagKey: to.StringPtr("shared"),
+			},
+		},
+	}
+
+	m, ok := results.Data.([]any)
+	if !ok {
```

Comment:
It would be good to add some debug logs here so we know there was something wrong with the data returned from the API.